pkg/scheduler/eventhandlers.go

```diff
@@ -456,12 +458,11 @@ func addAllEventHandlers(
 			}
 		case framework.ResourceClaim:
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
-				if handlerRegistration, err = informerFactory.Resource().V1alpha2().ResourceClaims().Informer().AddEventHandler(
-					buildEvtResHandler(at, framework.ResourceClaim, "ResourceClaim"),
-				); err != nil {
-					return err
-				}
-				handlers = append(handlers, handlerRegistration)
+				// No need to wait for this cache to be
+				// populated. If a claim is not yet in the
+				// cache, scheduling will get retried once it
+				// is.
+				resourceClaimCache.AddEventHandler(buildEvtResHandler(at, framework.ResourceClaim, "ResourceClaim"))
```

Comment:
We can make the`ResourceEventHandlerRegistration` part of the assumeCache and add to the handlerRegistration. Since the assumeCache is based on the informer, it makes sense to wait for the handler synced.